Firestore/core/test/unit/bundle/bundle_serializer_test.cc

```diff
@@ -1011,18 +1012,20 @@ TEST_F(BundleSerializerTest, DecodesStartAtCursor) {
   core::Query original =
       testutil::Query("colls")
           .AddingOrderBy(OrderBy("f1", "asc"))
-          .StartingAt(core::Bound::FromValue({Array("f1", 1000)},
-                                             /* is_before= */ true));
+          .StartingAt(core::Bound::FromValue(
+              SharedMessage<google_firestore_v1_ArrayValue>{Array("f1", 1000)},
```

Comment:
For cases like this, the compiler would have been able to figure out the template type if it were a function, not a class (in C++11; in C++17 class template arguments can be deduced as well). This is basically the reason for functions like `std::make_pair`. Consider adding a free function like `MakeSharedMessage` -- then this could likely be:
```cpp
MakeSharedMessage(Array("f1", 1000))
```
(applies throughout)

Reply:
Done. I was wondering why I needed to put the class parameter everywhere. Thanks for the cleanup suggestion.